src/vanilla.ts

```diff
@@ -109,6 +120,6 @@ export default function create<TState extends State>(
 
   const destroy: Destroy = () => listeners.clear()
   const api = { setState, getState, subscribe, destroy }
-  state = createState(setState, getState, api)
+  state = createState(setState as CustomSetState, getState, api)
```

Comment:
why is this type assertion necessary  now please?

Reply:
TS complains. I wonder if you have a better idea...

Reply:
Now, I realized this doesn't solve anything...

Reply:
```suggestion
  state = createState(setState, getState, api)
```
reverting this for now.